app/src/ui/lib/join.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Insert a separator element between each entry of an array of elements
+ *
+ * @param elements - A list of components
+ * @param separator - A string or component that will be placed between every element
+ */
+export function join(
```

Comment:
I believe we could simplify this method and make it generic at the same time by leveraging the index parameter in `reduce`

```ts
export function join<T, T1>(arr: T[], separator: T1): ReadonlyArray<T | T1> {
  return arr.reduce(
    (prev: Array<T | T1>, current: T, ix: number) =>
      ix === 0 ? [current] : [...prev, separator, current],
    []
  )
}
```

I feel the need to point out that we should only use this method for small arrays as the cost will increase quite dramatically for larger arrays.

I worry though that using something like this for JSX will mean that we'll have duplicated separators which is fine for strings but not so much for JSX elements.